### PR TITLE
Update onboarding docs

### PR DESCRIPTION
**Pull request by ThomasBrady**
# Description

* Adds contribution instructions (perhaps they would be better in an internal document rather than in the repo)
* Updates macos installation (i.e. how to get `clang-format-12`)
* Clarifies tracy setup 

# Checklist
- [ ] Reviewed the [contributing](https://github.com/stellar/stellar-core/blob/master/CONTRIBUTING.md#submitting-changes) document
- [ ] Rebased on top of master (no merge commits)
- [ ] Ran `clang-format` v8.0.0 (via `make format` or the Visual Studio extension)
- [ ] Compiles
- [ ] Ran all tests
- [ ] If change impacts performance, include supporting evidence per the [performance document](https://github.com/stellar/stellar-core/blob/master/performance-eval/performance-eval.md)

### CODE REVIEW

File: CONTRIBUTING.md
Comment (SirTyson) [inline CONTRIBUTING.md:19]: I think this is reversed. Usually, the original forked repo is refereed to as upstream. I think we should change this so that you clone your fork as origin and set the original repo as upstream. We can change the rebase command to `git fetch upstream && git rebase upstream/master`

File: INSTALL.md
Comment (SirTyson) [inline INSTALL.md:123]: Might want to also say that other versions of clang-format will not work properly in the note.

File: INSTALL.md
Comment (SirTyson) [inline INSTALL.md:126]: Instead of descriptions with example commands, it might be easier just to have a block that people can just copy paste. You essentially already have that, just turn the first download step into a `curl` and get rid of the words.

File: INSTALL.md
Comment (SirTyson) [inline INSTALL.md:175]: Nice! I had forgotten how long I battled with tracy versioning...

**Review by SirTyson — APPROVED** [review]
Thanks, I had forgotten I got hung up on some of these issues too. A few minor nitpicks.

File: INSTALL.md
Comment (SirTyson) [inline INSTALL.md:127]: Nit: I don't think we need this to be a one liner, and it would be a lot more readable if this was in a code block instead of inlined. We also don't need the cd:
```
curl -OL https://github.com/llvm/llvm-project/releases/download/llvmorg-12.0.0/clang+llvm-12.0.0-x86_64-apple-darwin.tar.xz 
tar -xf clang+llvm-12.0.0-x86_64-apple-darwin.tar.xz 
sudo cp -R clang+llvm-12.0.0-x86_64-apple-darwin/* bin/clang-format /usr/local/bin
```

**Review by SirTyson — APPROVED** [review]
LGTM, one small fix then we can merge this after a rebase.

File: CONTRIBUTING.md
Comment (marta-lokhova) [inline CONTRIBUTING.md:16]: I think it would be a bit nicer to keep this section focused on core-specific instructions, just to give contributors an idea on how to get started. So perhaps we don't need detailed git commands, and could simply mention that we use forked repos (as opposed to feature branches) for pull request flow, and generally keep our commit history tidy and up-to-date with master.

File: INSTALL.md
Comment (marta-lokhova) [inline INSTALL.md:122]: thanks for adding this!

**Review by marta-lokhova — COMMENTED** [review]
Nice changes! Just one suggestion

File: CONTRIBUTING.md
Comment (ThomasBrady) [reply CONTRIBUTING.md:16]: Fair, I think we would benefit from an explicit workflow but I agree it is verbose. I could modify this section to just say:

## Submitting Changes

Fork the `stellar/stellar-core` repo on github and submit a pull request from a branch in your fork. 
Key things to keep in mind:
* Keep your branch up-to-date by rebasing on `stellar/stellar-core` `master`
* Ensure your branch focuses on a single issue at a time
* Before merging, squash your commits and ensure you have a descriptive commit message

**Comment by marta-lokhova** [comment]
r+ 173526e045c2bef6645191e44423a8f7643e3d8f

**Pull request merged**
